linera-chain/src/data_types.rs

```diff
@@ -515,11 +593,9 @@ impl<'a> LiteCertificate<'a> {
         round: Round,
         mut signatures: Vec<(ValidatorName, Signature)>,
     ) -> Self {
-        if !is_strictly_ordered(&signatures) {
-            // Not enforcing no duplicates, check the documentation for is_strictly_ordered
-            // It's the responsibility of the caller to make sure signatures has no duplicates
-            signatures.sort_by_key(|&(validator_name, _)| validator_name)
-        }
+        // Not enforcing no duplicates, check the documentation for is_strictly_ordered
+        // It's the responsibility of the caller to make sure signatures has no duplicates
+        signatures.sort_by_key(|&(validator_name, _)| validator_name);
```

Comment:
(Good catch, Rust's sorting algorithm is already linear for sorted slices. :+1:)